app/src/main/java/io/homeassistant/companion/android/notifications/MessagingManager.kt

```diff
@@ -1233,6 +1242,34 @@ class MessagingManager @Inject constructor(
             return@withContext image
         }
 
+    private suspend fun saveTempAnimatedImage(serverId: Int, url: URL?, requiresAuth: Boolean = false): Uri? =
+        withContext(
+            Dispatchers.IO
+        ) {
+            if (url == null || url.path.endsWith("gif").not()) {
+                return@withContext null
+            }
+
+            val file = File(context.externalCacheDir, "animated_notification_image.gif")
```

Comment:
Using the same file name every time will result in overwriting any previous image. When someone sends two notifications with different gifs, won't this result in it showing the second gif for both notifications?

Reply:
I was doing that on purpose to alleviate concerns of using too much storage. While implementing I thought that notifications will be handled synchronously so the notification itself will have loaded the image data by the time the next notification comes through. 

Looking at it now and we are using coroutines so there could be a race condition. What would the preferred way of handling images be so that we do not use too much storage on these images? Perhaps remove any images over 1 hour old?

Reply:
Another thought: does this need to be external instead of the internal cache dir? Per the documentation internal will be automatically cleared by the system and is not visible to other apps, unlike external.

Reply:
> Perhaps remove any images over 1 hour old?

Wouldn't that result in non-working images if you view the notification >1 hour after receiving it?

Reply:
It does need to be external because otherwise the system is unable to load the file. I tried that first actually but it was failing with file not found.

To be clear,`FileProvider.getUriForFile(context, "${context.packageName}.provider", file)` came at the end and is definitely required. If this works with internal directories then that could be used, but I was assuming it would not

Reply:
> Wouldn't that result in non-working images if you view the notification >1 hour after receiving it?

my assumption is when the notification is delivered to the device and displayed for the user is when the system reads the image from storage, so in that case I don't think it would be an issue.

If the system doesn't load it until the user actually expands the notification then yeah that would be a problem.

Reply:
I guess we should test it before continuing this discussion, otherwise changes will be made where none are required and your code works perfectly fine

Reply:
I am pulling things up now to test if we can use internal storage. While doing that I will also test multiple notifications

Reply:
Okay, so my findings are that we must use an external directory, otherwise the `getUriForFile` fails. 

Also, there is a race condition here when multiple notifications are sent simultaneously, however event sent a second apart works as expected. I have pushed a commit to keep the files but delete previous images older than an hour.

Reply:
extra note to clarify, I did test the case where the screen is off and the screen is turned on after notifications are delivered

Reply:
So while this seems to work initiallly, after fixing deleting (see other comment) at some point the system seems to no longer have the gif in memory and if it was deleted from the directory it's gone, even though you still have the notification. Note the thumbnail still shows up (because that is a Bitmap attached to the notification itself), expanding doesn't show the image anymore:
![Closed notification with thumbnail](https://github.com/home-assistant/android/assets/8148535/fba6eba4-5db6-4803-b0d5-9e8a9da63d02)
![Opened notification with no image](https://github.com/home-assistant/android/assets/8148535/34498002-ea3d-4d14-bdd1-7b97cfedf161)

Quick steps to reproduce: set the cutoff time to 1 minute, send a notification, wait 1 minute, send another notification, and toggle dark mode on/off to force the system to reload stuff without waiting.